handlers/providers/azure.go

```diff
@@ -0,0 +1,54 @@
+package providers
+
+import (
+	b64 "encoding/base64"
+	"os"
+	"path/filepath"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/rancherio/go-rancher/client"
+)
+
+func init() {
+	if err := RegisterProvider("azure", azureHandler); err != nil {
+		log.Fatal("could not register azure provider")
+	}
+}
+
+func azureHandler(machine *client.Machine, machineDir string) error {
+	var data *string
+	var filename string
+	if machine.AzureConfig.SubscriptionCert != "" {
+		data = &machine.AzureConfig.SubscriptionCert
+		filename = "subscription-cert.pem"
+	} else {
+		data = &machine.AzureConfig.PublishSettingsFile
+		filename = "publish-settings.xml"
+	}
+	path, err := saveDataToFile(filename, *data, machineDir)
+	if err != nil {
+		return err
+	}
+	*data = path
```

Comment:
For the record, discussed offline, this doesnt change the value store in cattle, its jsut used so that the command can be constructed properly locally. We can live it as it is.